spec/ios_download_strings_files_from_glotpress_spec.rb

```diff
@@ -55,16 +54,127 @@ def test_gp_download(filters:, tablename:, expected_gp_params:)
     it 'uses a custom table name for the `.strings` files if provided' do
       test_gp_download(
         filters: nil,
-        tablename: "MyApp",
+        tablename: 'MyApp',
         expected_gp_params: { 'filters[status]': 'current', format: 'strings' }
       )
     end
   end
 
   describe 'error handling' do
-    it 'shows an error if an invalid locale is provided (404)'
-    it 'shows an error if the file cannot be written in the destination'
-    it 'reports if a downloaded file is invalid by default'
-    it 'does not report invalid downloaded files if skip_file_validation:true'
+    it 'shows an error if an invalid locale is provided (404)' do
+      Dir.mktmpdir('a8c-release-toolkit-tests-') do |tmp_dir|
+        # Arrange
+        stub = gp_stub(locale: 'unknown-locale', query: { 'filters[status]': 'current', format: 'strings' }).to_return(status: [404, 'Not Found'])
+        error_messages = []
+        allow(FastlaneCore::UI).to receive(:error) { |message| error_messages.append(message) }
+
+        # Act
+        run_described_fastlane_action(
+          project_url: gp_fake_url,
+          locales: { 'unknown-locale': 'Base' },
+          download_dir: tmp_dir
+        )
+
+        # Assert
+        expect(stub).to have_been_made.once
+        file = File.join(tmp_dir, 'Base.lproj', 'Localizable.strings')
+        expect(File).not_to exist(file)
+        expect(error_messages).to eq(['Error downloading locale `unknown-locale` — 404 Not Found'])
```

Comment:
Nitpick: Since `file` is only used one, inlining it wouldn't make the code less DRY and also result in a nice `expect` only assert section.

```suggestion
        expect(stub).to have_been_made.once
        expect(File).not_to exist(File.join(tmp_dir, 'Base.lproj', 'Localizable.strings'))
        expect(error_messages).to eq(['Error downloading locale `unknown-locale` — 404 Not Found'])
```

Not a big deal at all, happy for it to stay as it is. Just throwing it here in case you like it too...

Reply:
Addressed in fd678842e1ef8a6fe3ffbd33bd403f1354b420e5